packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -116,6 +116,35 @@ class _DropdownMenuItemButton<T> extends StatefulWidget {
 }
 
 class _DropdownMenuItemButtonState<T> extends State<_DropdownMenuItemButton<T>> {
+  late final CurvedAnimation _fadeOpacity;
+
+  @override
+  void initState() {
+    super.initState();
+
+    final double unit = 0.5 / (widget.route.items.length + 1.5);
+
+    if (widget.itemIndex == widget.route.selectedIndex) {
```

Comment:
this needs a test that verifies that the widget works correctly even if `widget.itemIndex` is changed and another that tests that it works when `widget.route` changes.

Reply:
@Hixie, thanks a lot for the review.

I changed `CurvedAnimation` for `CurvedTween` to avoid disposal need. 
Could you please take a look at the updated code?

Reply:
The updated code seems semantically identical.

It still needs a test that verifies that the widget works correctly even if `widget.itemIndex` is changed and another that tests that it works when `widget.route` changes, though.

Reply:
I'm sorry for the silence from my side.
Thanks a lot for taking a look at this PR again.

I checked the source code of the `DropdownButton` widget and figured out that:
1) `_dropdownRoute` (`_DropdownRoute`) gets created only when we tap on the `DropdownButton`;
https://github.com/flutter/flutter/blob/8e3b7a9cb5d1e4558d97fa6024ceff17986903ca/packages/flutter/lib/src/material/dropdown.dart#L1342-L1356
2) when the dropdown menu is shown on top of the button (the pushed route is not removed), the new `_DropdownRoute` won't be pushed;
3) changes to the `DropdownButton` widget (e.g. updating of `_selectedIndex`) do not affect or update `_dropdownRoute` (it seems like this is intended by the creators of the `DropdownButton` widget).

Could you please help me to figure out how can I test the scenarios of route/selectedIndex change in `DropdownButton` according to the previous statements?

Reply:
@Hixie , you mean adding tests for _DropdownMenuItemButton, not for DropdownButton, right? 

_DropdownMenuItemButton is private. In order to test it directly, we need to make it public and @visibleForTesting.
That means we will need to add all doc comments. Doc comments will not hurt. 
But: are you sure direct testing of _DropdownMenuItemButton worth it, knowing public users of it are well test covered?

Reply:
@HansMuller , while Hixie is on vacation, can you review this fix?